src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -109,8 +106,16 @@ class SchemaService(private val options: Neo4jOptions, private val jobId: String
     }
 
     structFields ++= (try {
-        val query = "CALL apoc.meta.relTypeProperties({ includeRels: $rels })"
-        val params = Map[String, AnyRef]("rels" -> Seq(options.relationshipMetadata.relationshipType).asJava)
+        val query =
+          """CALL apoc.meta.relTypeProperties({ includeRels: $rels }) YIELD sourceNodeLabels, targetNodeLabels,
+            | propertyName, propertyTypes
+            |WITH *
+            |WHERE sourceNodeLabels = $sourceLabels AND targetNodeLabels = $targetLabels
```

Comment:
shouldn't we rather pass this to the proc? seems a big waste. If it's based on apoc.meta.* it should support include and exclude functionality.

Reply:
Unfortunately at this moment, the procedure does not take those parameters as configuration params

Reply:
oh ok, that's really unfortunate.

Reply:
But we can change that.

Reply:
yes